README.md

```diff
@@ -1,2 +1,289 @@
-# distributed-press-cli
-Command Line Interface for the Distributed Press API and Social Inbox
+# Distributed Press CLI
+
+A **Command Line Interface (CLI)** for interacting with the Distributed Press API and Social Inbox. Manage your sites, register your ActivityPub actor, and engage with followers seamlessly from the command line.
+
+## Table of Contents
+
+- [Installation](#installation)
+- [Setup](#setup)
+  - [Generate a Keypair](#generate-a-keypair)
+  - [Set Your Authentication Token](#set-your-authentication-token)
+  - [Register Your Actor](#register-your-actor)
+- [Configuration](#configuration)
+- [Usage](#usage)
+  - [Send a Post to Followers](#send-a-post-to-followers)
+- [Commands](#commands)
+  - [`generate-keypair`](#generate-keypair)
+  - [`set-auth-token`](#set-auth-token)
+  - [`register-actor`](#register-actor)
+  - [`send-post`](#send-post)
+- [Examples](#examples)
+- [License](#license)
+- [Resources](#resources)
+
+## Installation
+
+### Using npx
+
+You can use `dp-cli` without installation by running:
+
+```bash
+npx dp-cli <command>
+```
+
+### Global Installation with npm
+
+To install `dp-cli` globally, use:
+
+```bash
+npm install -g dp-cli
+```
+
+Once installed, you can use `dp-cli` from anywhere in your terminal.
+
+## Setup
+
+Before using the CLI, you need to generate a keypair, set your authentication token, and register your ActivityPub actor.
+
+### Generate a Keypair
+
+Run the following command to generate a new RSA keypair:
```

Comment:
Maybe add an explainer saying this is for authenticating in the social inbox?

Can this be skipped if we only use the DP API for static file publishing? Might want to make that clearer.